missing _mm_crc32_u64 on 32 bit build

On an Ubuntu 14.04.2 32 bit system with gcc 4.8.2, I get the following build error:

```
util/crc32c.cc: In function ‘void rocksdb::crc32c::Fast_CRC32(uint64_t*, const uint8_t**)’:
util/crc32c.cc:318:39: error: ‘_mm_crc32_u64’ was not declared in this scope
   *l = _mm_crc32_u64(*l, LE_LOAD64(*p));
```

I don't know if it is supposed to exist on a 32 bit system or not, but `__SSE4_2__` is defined, so the code assumes it exists.

--- a/util/crc32c.cc
+++ b/util/crc32c.cc
@@ -292,9 +292,11 @@ static inline uint32_t LE_LOAD32(const uint8_t *p) {
 }
 
 #ifdef __SSE4_2__
+#ifdef __LP64__
 static inline uint64_t LE_LOAD64(const uint8_t *p) {
   return DecodeFixed64(reinterpret_cast<const char*>(p));
 }
+#endif
 #endif
 
 static inline void Slow_CRC32(uint64_t* l, uint8_t const **p) {

--- a/util/crc32c.cc
+++ b/util/crc32c.cc
@@ -319,9 +319,9 @@ static inline void Fast_CRC32(uint64_t* l, uint8_t const **p) {
   *l = _mm_crc32_u64(*l, LE_LOAD64(*p));
   *p += 8;
 #else
-  *l = _mm_crc32_u32(*l, LE_LOAD32(*p));
+  *l = _mm_crc32_u32((unsigned int)*l, LE_LOAD32(*p));
   *p += 4;
-  *l = _mm_crc32_u32(*l, LE_LOAD32(*p));
+  *l = _mm_crc32_u32((unsigned int)*l, LE_LOAD32(*p));
   *p += 4;
 #endif
 #else

--- a/util/crc32c.cc
+++ b/util/crc32c.cc
@@ -319,9 +319,9 @@ static inline void Fast_CRC32(uint64_t* l, uint8_t const **p) {
   *l = _mm_crc32_u64(*l, LE_LOAD64(*p));
   *p += 8;
 #else
-  *l = _mm_crc32_u32((unsigned int)*l, LE_LOAD32(*p));
+  *l = _mm_crc32_u32(static_cast<unsigned int>(*l), LE_LOAD32(*p));
   *p += 4;
-  *l = _mm_crc32_u32((unsigned int)*l, LE_LOAD32(*p));
+  *l = _mm_crc32_u32(static_cast<unsigned int>(*l), LE_LOAD32(*p));
   *p += 4;
 #endif
 #else

--- a/util/crc32c.cc
+++ b/util/crc32c.cc
@@ -315,8 +315,15 @@ static inline void Slow_CRC32(uint64_t* l, uint8_t const **p) {
 
 static inline void Fast_CRC32(uint64_t* l, uint8_t const **p) {
 #ifdef __SSE4_2__
+#ifdef __LP64__
   *l = _mm_crc32_u64(*l, LE_LOAD64(*p));
   *p += 8;
+#else
+  *l = _mm_crc32_u32(*l, LE_LOAD32(*p));
+  *p += 4;
+  *l = _mm_crc32_u32(*l, LE_LOAD32(*p));
+  *p += 4;
+#endif
 #else
   Slow_CRC32(l, p);
 #endif

